packages/styled/types/tests-base.tsx

```diff
@@ -432,3 +432,13 @@ const Section = styled('section')`
 `
 ;<StyledWithAs as="section" />
 ;<StyledWithAs as={Section} />
+
+{
+  const MemoedComponent = React.memo((_props: { foo: string }) => {
```

Comment:
this is not relevant for this change, just something I've wanted to check - there were no other types tests involving `.memo` so I've decided to add this here.